docs/root/faq/edge.rst

```diff
@@ -0,0 +1,85 @@
+.. _faq_edge:
+
+Can I use Envoy as an edge proxy?
+=================================
+
+Envoy is a production-ready edge proxy, and performs well in that role, however,
+the default settings are tailored for the service mesh use case, and some values
+need to be adjusted when using Envoy as an edge proxy.
+
+TCP proxies should configure: :ref:`overload_manager <config_overload_manager>`,
+:ref:`listener buffer limits <envoy_api_field_Listener.per_connection_buffer_limit_bytes>` to 32,768 bytes
+and :ref:`cluster buffer limits <envoy_api_field_Cluster.per_connection_buffer_limit_bytes>` to 32,768 bytes.
+
+HTTP proxies should additionally configure:
+:ref:`use_remote_address <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.use_remote_address>` to true,
+and HTTP/2 settings (e.g.
+:ref:`HTTP/2 maximum concurrent streams limit <envoy_api_field_core.Http2ProtocolOptions.max_concurrent_streams>` to 100,
+:ref:`HTTP/2 initial stream window size limit <envoy_api_field_core.Http2ProtocolOptions.initial_stream_window_size>` to 65,535 bytes,
+and :ref:`HTTP/2 initial connection window size limit <envoy_api_field_core.Http2ProtocolOptions.initial_connection_window_size>` to 1,048,576 bytes).
+
+The following is a YAML example of the above recommendation.
+
+.. code-block:: yaml
+
+  overload_manager:
+    refresh_interval:
+      seconds: 0
+      nanos: 250000000 # 0.25s
+    resource_monitors:
+    - name: "envoy.resource_monitors.fixed_heap"
+      config:
+        # TODO: Tune for your system.
+        max_heap_size_bytes: 2147483648 # 2GB
+    actions:
+    - name: "envoy.overload_actions.stop_accepting_requests"
+      triggers:
+      - name: "envoy.resource_monitors.fixed_heap"
+        threshold:
+          value: 0.95
```

Comment:
@eziskind can you take a look at these settings?

Reply:
Maybe also add an overload action to try shrink the heap when memory pressure is high. This is needed because tcmalloc doesn't automatically release memory back to the OS (see discussion here https://groups.google.com/d/topic/envoy-dev/DrtnkaO9VRU/discussion).

To configure add a block like this to the "actions:" section:
- name: "envoy.overload_actions.shrink_heap"
      triggers:
        - name: "envoy.resource_monitors.fixed_heap"
          threshold:
            value: 0.95

Reply:
Added, thanks!